### PR TITLE
devguide: bring contribution process page from redmine - v3

### PR DESCRIPTION
**Pull request by jufajardini**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4929

Previous PR: https://github.com/OISF/suricata/pull/7373

Describe changes:
Last PR: 
- remove outdated reference
- address nit comment
- indicate that we usually have 2 or 3 branches
- incorporate suggestions, reword and reword part about expectations and - community-supported features

This PR:
- added some guidelines about documentation writing: line length and images/diagrams preferences.

### CODE REVIEW

File: doc/userguide/devguide/codebase/contributing/contribution-process.rst [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+They are also important if you would like to see your new feature officially
+added to our tool: the ticket documents your ideas so  we can analyze how do they
+fit in our plans for Suricata, and, if the feature is accepted, we can properly
+track progress etc...
+
+For really trivial fixes or cleanups we won't need that.
+
+Assign the ticket to yourself. For this you will need to have the "developer"
+role. Please get in touch with us to request this role.
```
Comment (victorjulien) [inline doc/userguide/devguide/codebase/contributing/contribution-process.rst:24]: should this have a link for methods of getting in touch?

File: doc/userguide/devguide/codebase/contributing/contribution-process.rst [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+The team and/or community members can then consider offering help. It is best 
+to indicate this prior to doing the actual work, because we will reject features
+if no one steps up.
+
+It is also important to note that ``community supported`` features  will be
+disabled by default, and if it brings in new dependencies (libraries or Rust
+crates) those will also be optional and disabled by default.
+
+**Supporting a feature** means to actually *maintain* it:
```
Comment (victorjulien) [inline doc/userguide/devguide/codebase/contributing/contribution-process.rst:75]: some end user support in forum and/or chat?

File: doc/userguide/devguide/codebase/contributing/contribution-process.rst [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+to improve GeoIP? Name your branch "geoip-feature-123-v1". The "-v1" addition 
+for feedback. When incorporating feedback you will have to create a new branch
+for each pull request. So when you address the first feedback, you will work in
+"geoip-feature-123-v2" and so on.
+
+More guidance on creating branches via:
+
+GitHub: `Creating a branch to do your changes <https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Github_work_flow#Creating-a-branch-to-do-your-changes>`_
+
+Git: `Create your branch <https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Git_work_flow#Create-your-branch>`_
```
Comment (victorjulien) [inline doc/userguide/devguide/codebase/contributing/contribution-process.rst:116]: remove?

File: doc/userguide/devguide/codebase/contributing/contribution-process.rst [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+It's useful to create descriptive branch names. If you're working on ticket 123
+to improve GeoIP? Name your branch "geoip-feature-123-v1". The "-v1" addition 
+for feedback. When incorporating feedback you will have to create a new branch
+for each pull request. So when you address the first feedback, you will work in
+"geoip-feature-123-v2" and so on.
+
+More guidance on creating branches via:
+
+GitHub: `Creating a branch to do your changes <https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Github_work_flow#Creating-a-branch-to-do-your-changes>`_
```
Comment (victorjulien) [inline doc/userguide/devguide/codebase/contributing/contribution-process.rst:114]: ticket for also moving this to the dev docs?

File: doc/userguide/devguide/codebase/contributing/contribution-process.rst [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+Git: `Create your branch <https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Git_work_flow#Create-your-branch>`_
+
+
+Coding Style
+============
+
+We have a :ref:`Coding Style` that must be followed.
+
+Documentation Style
```
Comment (victorjulien) [inline doc/userguide/devguide/codebase/contributing/contribution-process.rst:124]: perhaps mention that we want things to look good in pdf and readthedocs

File: doc/userguide/devguide/codebase/contributing/contribution-process.rst [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+Send a pull request
+===================
+
+The pull request is used to request inclusion of your patches into the main 
+repository. Before it is merged, it will be reviewed and pushed through a QA
+process.
+
+Please consider our :ref:`Pull Requests Criteria` when submitting.
+
+On GitHub we have 'travis-ci' integration enabled. This means some automated
```
Comment (victorjulien) [inline doc/userguide/devguide/codebase/contributing/contribution-process.rst:150]: github-ci these days

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/7546

**Pull request closed without merging**